src/renderers/dom/client/__tests__/ReactMount-test.js

```diff
@@ -163,6 +163,27 @@ describe('ReactMount', function() {
     );
   });
 
+  it('should warn when component does not follow proper case convention',
+    function() {
+      var container = document.createElement('container');
+      var myDiv = <div>React</div>;
+      var mySvg = <svg><polygon points="-145.6 556.9 104.8-145.6 429"/></svg>;
+      var centered = React.createClass({
```

Comment:
I added the `createClass`line to test that the warning was actually triggering but yes feel free to delete it.

> This is effectively just catching unknown elements

Correct as long as the `element.type` passes the criteria:
```js
if (element == null || typeof element.type !== 'string' ||
    element.type.match(/\-|\s+/) !== null ||
    ReactDOMFactories.hasOwnProperty(element.type)) {
    return;
  }
```